### PR TITLE
Yuanhenglee-proposal-submission

### PR DESCRIPTION
**Pull request by yuanhenglee**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please get rid of unneeded changesets, e.g., `6234b0d8269cc75afd99e50a3f1dacbe96af72c9`, from history.  You may use rebase.

I will review once you correct the history.  Please leave a comment when you do.

**Comment by yuanhenglee** [comment]
> Please get rid of unneeded changesets, e.g., `6234b0d8269cc75afd99e50a3f1dacbe96af72c9`, from history. You may use rebase.
> 
> I will review once you correct the history. Please leave a comment when you do.

Thank you for pointing that out. I have done correcting the old commits.

File: project/yuanhenglee/README.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# BasiCPP Pitch: A C++ implementation for AMT(Automatic Music Transcription)
```
Comment (yungyuc) [inline project/yuanhenglee/README.rst:1]: You name the file as rst but use markdown in the content.  It's better to make them consistent.

File: project/yuanhenglee/README.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+## Prospective Users
+
+Music producers and engineers can employ it to transcribe audio recordings into
+MIDI files, facilitating tasks like remixing and editing. Music researchers and
+analysts can conduct in-depth compositional studies, while educators and
+students can use it for ear training and composition exercises.
+
+## System Architecture
+
+![image](https://github.com/yuanhenglee/basicpp-pitch/blob/master/pics/NSD_project_flowchart.drawio.png)
```
Comment (yungyuc) [inline project/yuanhenglee/README.rst:48]: I am not seeing the picture.

File: project/yuanhenglee/README.rst [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+Music Transcription) library written in C++.
+
+## Basic Information
+
+BasiCPP Pitch is an instrument-agnostic and polyphonic-capable AMT (Automatic
+Music Transcription) library written in C++.
+
+- github repository:[repo](https://github.com/yuanhenglee/basicpp-pitch)
+
+Provide any compatible audio file, the library will generate a MIDI file with
```
Comment (yungyuc) [inline project/yuanhenglee/README.rst:13]: It won't be hard to make beautiful slides using the waveform analysis and notes.

File: project/yuanhenglee/README.rst [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+analysts can conduct in-depth compositional studies, while educators and
+students can use it for ear training and composition exercises.
+
+## System Architecture
+
+![image](https://github.com/yuanhenglee/basicpp-pitch/blob/master/pics/NSD_project_flowchart.drawio.png)
+
+### Input
+
+The system takes an audio file that needs to be transcribed. as input.
```
Comment (yungyuc) [inline project/yuanhenglee/README.rst:52]: How many analytical approaches do you want to implement in the project time frame?

File: project/yuanhenglee/README.rst [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+1. Automatic build system: `CMake`
+2. Version control: `Git`
+3. Testing framework: `Pytest`
+4. Documentation: `README`, `PyDoc`
+
+## Schedule
+
+* Week 1 (10/30):
+    - [ ] Set up the environment and project structure
+    - [ ] Test the pure Python prototype
```
Comment (yungyuc) [inline project/yuanhenglee/README.rst:145]: Do you have the Python prototype ready now?

**Review by yungyuc — CHANGES_REQUESTED** [review]
The structure looks complete.  You seem to have thought about it clearly.  It may potentially be an interesting project.

File: project/yuanhenglee/README.rst (hunk 1 above)
Comment (yuanhenglee) [reply project/yuanhenglee/README.rst]: Sure! I suppose the picture issue below will be fixed after I correct the file into markdown.

File: project/yuanhenglee/README.rst (hunk 4 above)
Comment (yuanhenglee) [reply project/yuanhenglee/README.rst]: By analytical approaches, does that mean how many modules do I have to implement to accomplish the problem?
If that is the case, the primary two will be CQT and the inference module for the CNN.
I'm sorry if I misunderstood the question.

File: project/yuanhenglee/README.rst (hunk 5 above)
Comment (yuanhenglee) [reply project/yuanhenglee/README.rst]: The Python prototype will be [the one](https://github.com/spotify/basic-pitch/tree/main) provided by Spotify. I think the process will be pretty much the same except that basic-pitch uses TensorFlow and Librosa, which I'm going to rewrite in C++.

**Comment by yungyuc** [comment]
When you are ready for the next review, please leave a global comment to request for it.

**Pull request merged**